charts/posthog/templates/_helpers.tpl

```diff
@@ -171,66 +162,97 @@ Set pgbouncer URL
 {{- end -}}
 
+{*
+   ------ REDIS ------
+*}
+
 {{/*
-Set redis host
+Return the Redis fullname
 */}}
-{{- define "posthog.redis.host" -}}
-{{- if .Values.redis.enabled -}}
-{{- template "posthog.redis.fullname" . -}}-master
+{{- define "posthog.redis.fullname" -}}
+{{- if .Values.redis.fullnameOverride -}}
+{{- .Values.redis.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else if .Values.redis.nameOverride -}}
+{{- printf "%s-%s" .Release.Name .Values.redis.nameOverride | trunc 63 | trimSuffix "-" -}}
 {{- else -}}
-{{- .Values.redis.host | quote -}}
+{{- printf "%s-%s" (include "posthog.fullname" .) "redis" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Set redis secret
+Return the Redis host
 */}}
-{{- define "posthog.redis.secret" -}}
-{{- if .Values.redis.enabled -}}
-{{- template "posthog.redis.fullname" . -}}
+{{- define "posthog.redis.host" -}}
+{{- if .Values.redis.enabled }}
+    {{- printf "%s-master" (include "posthog.redis.fullname" .) -}}
 {{- else -}}
-{{- template "posthog.fullname" . -}}
+    {{- printf "%s" .Values.externalRedis.host -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Set redis secretKey
+Return the Redis port
 */}}
-{{- define "posthog.redis.secretKey" -}}
-{{- if .Values.redis.enabled -}}
-"redis-password"
+{{- define "posthog.redis.port" -}}
+{{- if .Values.redis.enabled }}
+    {{- printf "6379" | quote -}}
 {{- else -}}
-{{- default "redis-password" .Values.redis.existingSecretKey | quote -}}
+    {{- .Values.externalRedis.port | quote -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Set redis password
+Return true if a secret object for Redis should be created
 */}}
-{{- define "posthog.redis.password" -}}
-{{ .Values.redis.password | default "" }}
+{{- define "posthog.redis.createSecret" -}}
+{{- if and (not .Values.redis.enabled) (not .Values.externalRedis.existingSecret) .Values.externalRedis.password }}
+    {{- true -}}
+{{- end -}}
 {{- end -}}
-
 
 {{/*
-Set redis password base64
+Return the Redis secret name
 */}}
-{{- define "posthog.redis.passwordb64" -}}
-{{ .Values.redis.password | default "" | b64enc | quote }}
+{{- define "posthog.redis.secretName" -}}
+{{- if .Values.redis.enabled }}
+    {{- if .Values.redis.auth.existingSecret }}
+        {{- printf "%s" .Values.redis.auth.existingSecret -}}
+    {{- else -}}
+        {{- printf "%s" (include "posthog.redis.fullname" .) -}}
+    {{- end -}}
+{{- else if .Values.externalRedis.existingSecret }}
+    {{- printf "%s" .Values.externalRedis.existingSecret -}}
+{{- else -}}
+    {{- printf "%s-external" (include "posthog.redis.fullname" .) -}}
+{{- end -}}
 {{- end -}}
-
 
 {{/*
-Set redis port
+Return the Redis secret key
 */}}
-{{- define "posthog.redis.port" -}}
-{{- if .Values.redis.enabled -}}
-    6379
+{{- define "posthog.redis.secretPasswordKey" -}}
```

Comment:
We do a clever trick here - checking for keys existing. Should we do the same trick to validate that users aren't passing in old, removed keys (redis.host, etc)?

Reply:
We had a similar concern for the PR that deprecated custom `StorageClass`es and like there I'm inclined to don't add any additional check and simply provide detailed information in the release notes.

Reply:
I think it's worth revisting this: people will skip reading release notes or miss things they need to update, which in turn will translate to a broken experience and more support burden for us.

We could create a single block somewhere which for every removed value gives some guidance that this is now removed. This shouldn't be too much of a maintanence burden but would get us most of the way there in terms of UX.

Reply:
> people will skip reading release notes or miss things they need to update, which in turn will translate to a broken experience and more support burden for us.

I agree but in that case Helm should automatically rollback to the previous version. Also the new pods will not pass the readiness check so also in that case no downtime / no broken experience is expected.

Reply:
> I agree but in that case Helm should automatically rollback to the previous version

Can we guarantee that happens every time though? In any case, PRs over issues - if you are willing to delay the major version release a bit, will have a PR for you soon :)

Reply:
https://github.com/PostHog/charts-clickhouse/pull/261